### PR TITLE
Proper erasing of characters - Fixes issue #51

### PR DESCRIPTION
**Pull request by chaitan94**
Have tested this on Arch, Debian and Ubuntu - zsh, bash and plain sh.

### CODE REVIEW

File: spinner.go [hunk 1]
```diff
@@ -297,7 +298,8 @@ func (s *Spinner) UpdateCharSet(cs []string) {
 // Caller must already hold s.lock.
 func (s *Spinner) erase() {
 	n := utf8.RuneCountInString(s.lastOutput)
-	for _, c := range []string{"\b", " ", "\b"} {
+	del, _ := hex.DecodeString("7f")
```
Comment (briandowns) [inline spinner.go]: Is there a default value we can set on error maybe rather than ignoring it?

Comment (briandowns) [reply spinner.go]: Meh, we can sort that later. Thanks for this PR!

**Comment by chaitan94** [comment]
@briandowns 

Thanks for the merge. But unfortunately, I just noticed that while this works for pattern 39, it seems to cause issues for long variable-length patterns like 31 or if there is appended text. In this case the whole text doesn't get erased. (You can run the _example/main.go to understand this better)

In my previous testing I did not include such cases, so I seem to have missed it.

**Comment by briandowns** [comment]
Thanks for noticing that. Is there any chance you'd be able to submit another PR?

**Comment by chaitan94** [comment]
Unfortunately, I am traveling right now and won't be able to look into this, at least until Monday. If it isn't fixed by then, I will try to see if I can do anything.

**Pull request merged**